File: Leetcode/PY/Leetcode219-存在重复元素II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        st,n = set(),len(nums)
        for i in range(n):
            if i > k: st.remove(nums[i - k - 1])
            if nums[i] in st: return True
            st.add(nums[i])
        return False

File: Leetcode/JS/Leetcode219-存在重复元素II.js
var containsNearbyDuplicate = function(nums, k) {
    const st = new Set();
    const n = nums.length;
    for(let i = 0;i < n;i++) {
        if(i > k) st.delete(nums[i - k - 1]);
        if(st.has(nums[i])) return true;
        st.add(nums[i]);
    }
    return false;
};